### ISSUE TITLE
Include permalink in error message when replied to message fails to load

### ISSUE THREAD

**Issue by MadLittleMods** [vector-im/element-web#21536]
#### What would you like to do?

When a reply fails to load, include a permalink or at least `event_id` for the message it's trying to load so it's possible to reference and find the event in question.

Currently, the error message just says the following which has no details.

```
Unable to load event that was replied to, it either does not exist or you do not have permission to view it.
```

![](https://user-images.githubusercontent.com/558581/159744501-f068c37f-a8a1-4b39-ae2f-96ee47ebf3de.png)

#### Why would you like to do it?

It's not easy to find the message it's trying to reference yourself or debug the situation. Just have to View source and dive into the gritty details.

#### How would you like to achieve it?

Add a permalink to the event in question. It probably exists if someone is trying to reference it.

### Additional context

Related to:

 - https://github.com/vector-im/element-web/issues/19769

**Comment by t3chguy** [comment]
Given that its not loading under the /event/ API then it realistically won't load as a permalink either (/context/ API) so a permalink doesn't do anyone much good, surely?

**Comment by t3chguy** [comment]
I think solving that issue by the suggestion in that issue would be far better then...

> Looks like this is caused by the server being unhappy. We could probably intercept the calls that the replies make to pull from local cache first.

**Comment by MadLittleMods** [comment]
> Giving a user a link which we know has a significant chance of just erroring

I think this assumption is wrong. The link won't error in all cases. The reply could fail to load in many different ways, not just homeserver network issues.

Giving a user no detail on what went wrong or nudge on how to fix things on their own is a smell. Plus the current reason it gives is wrong for why it failed for me: "it either does not exist or you do not have permission to view it."

**Comment by t3chguy** [comment]
When would `/event/` fail when a permalink (`/context/`) function fine, especially once fixing the local cache suggested in the other issue?

**Comment by MadLittleMods** [comment]
@t3chguy It could fail in the JavaScript code too. As an example in my case, `/event/` didn't fail in my case yet the reply couldn't load.

If we're arguing that we should fix the JavaScript instead as well, yes! But any new regression can come in the future to cause replies to not load again. So there should be a escape hatch for users to workaround the problem in the mean time.

**Comment by t3chguy** [comment]
> It could fail in the JavaScript code too. As an example in my case, /event/ didn't fail in my case yet the reply couldn't load.

Do you have logs to back that up? I haven't seen any JS failures in this area and don't see any open bugs with logs about them, all current known issues are server/network issue related.

**Comment by MadLittleMods** [comment]
@t3chguy Yes, No error occurred though. I can even still reproduce so here is rageshake logs while staring at it, https://github.com/matrix-org/element-web-rageshakes/issues/11639

**Comment by t3chguy** [comment]
Turns out for reply fetching we still use the /context/ API rather than the /event/ one (/context/ is also used for fetching permalinks)

If no error was thrown then likely it wasn't a JS error, we rarely swallow errors.
Here we do swallow errors, but only if the /context/ fetch fails, would be good to log here regardless https://github.com/matrix-org/matrix-react-sdk/blob/3f67a389c148d661d024d0a0cfd8a9bc5f34e1f8/src/components/views/elements/ReplyChain.tsx#L169

Can you repro with your devtools open and observe the `/context/` request?

**Comment by MadLittleMods** [comment]
> Can you repro with your devtools open and observe the /context/ request?

@t3chguy Yes, 200 OK for `/context/`.

The place I can still reproduce is within a thread and replies and what not (around this sort of stuff https://github.com/vector-im/element-web/issues/21533) but the point still stands. Things go wrong in unexpected ways and this same sort of thing could happen in the main timeline for a message that should be there.

**Comment by t3chguy** [comment]
> Even after https://github.com/vector-im/element-web/issues/21533 is now merged and deployed

#21533 never claimed to fix it, nor did it even think it did

**Comment by MadLittleMods** [comment]
Before https://github.com/vector-im/element-web/issues/21533, it was unclear to me where the replies were meant to be (in main timeline or thread panel) and whether it was more of a undefined scenario since it was duplicated and would disappear after that PR. Now that is more settled and I can narrow down a bug more clearly.

**Comment by turt2live** [comment]
it might be possible to add a developer-only button of some kind if we're not certain it'd be safe for the general population. Like https://github.com/matrix-org/matrix-react-sdk/pull/7537 but for replies.
